operator/pkg/cli/migratev2/adminconsole.go

```diff
@@ -0,0 +1,157 @@
+
+	return nil
+}
+
+func updateAdminConsoleClusterConfig(ctx context.Context, cli client.Client) error {
+	var clusterConfig k0sv1beta1.ClusterConfig
+	nsn := apitypes.NamespacedName{Name: "k0s", Namespace: "kube-system"}
+	err := cli.Get(ctx, nsn, &clusterConfig)
+	if err != nil {
+		return fmt.Errorf("get k0s cluster config: %w", err)
+	}
+
+	for ix, ext := range clusterConfig.Spec.Extensions.Helm.Charts {
+		if ext.Name == "admin-console" {
+			values, err := updateAdminConsoleChartValues([]byte(ext.Values))
+			if err != nil {
+				return fmt.Errorf("update admin-console chart values: %w", err)
+			}
+			ext.Values = string(values)
+
+			clusterConfig.Spec.Extensions.Helm.Charts[ix] = ext
+		}
+	}
+
+	err = cli.Update(ctx, &clusterConfig)
+	if err != nil {
+		return fmt.Errorf("update k0s cluster config: %w", err)
+	}
+
+	return nil
+}
+
+func updateAdminConsoleChartValues(values []byte) ([]byte, error) {
+	var m map[string]interface{}
+	err := yaml.Unmarshal(values, &m)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshal values: %w", err)
+	}
+
+	m["isEC2Install"] = "true"
+
+	b, err := yaml.Marshal(m)
+	if err != nil {
+		return nil, fmt.Errorf("marshal values: %w", err)
+	}
+
+	return b, nil
+}
+
+// setIsEC2InstallInstallationStatus is needed to inform the operator that the installation has
+// been upgraded to v2 which prevents the operator from reconciling the installation.
+func setIsEC2InstallInstallationStatus(ctx context.Context, cli client.Client, in *ecv1beta1.Installation) error {
+	copy := in.DeepCopy()
+
+	if copy.Status.Conditions == nil {
+		copy.Status.Conditions = []metav1.Condition{}
+	}
+	copy.Status.SetCondition(metav1.Condition{
+		Type:   ConditionTypeIsEC2Install,
+		Status: metav1.ConditionTrue,
+		Reason: "MigrationComplete",
+	})
+
+	if err := cli.Status().Patch(ctx, copy, client.MergeFrom(in.DeepCopy())); err != nil {
+		return fmt.Errorf("patch crd installation status: %w", err)
+	}
+
+	return nil
+}
+
+// waitForAdminConsoleDeployment waits for the kotsadm pod to be updated as the service account
+// does not have permissions to get deployments.
+func waitForAdminConsoleDeployment(ctx context.Context, cli client.Client) error {
+	backoff := wait.Backoff{Steps: 60, Duration: 5 * time.Second, Factor: 1.0, Jitter: 0.1}
+	var lasterr error
+	if err := wait.ExponentialBackoffWithContext(
+		ctx, backoff, func(ctx context.Context) (bool, error) {
+			ready, err := isAdminConsoleDeploymentUpdated(ctx, cli)
+			if err != nil {
+				lasterr = fmt.Errorf("check deployment: %w", err)
+				return false, nil
+			}
+			return ready, nil
+		},
+	); err != nil {
+		if lasterr != nil {
+			return lasterr
+		}
+		return err
+	}
+	return nil
+}
+
+// isAdminConsoleDeploymentUpdated checks that the kotsadm pod has the desired environment
+// variable, as the service account does not have permissions to get deployments.
+func isAdminConsoleDeploymentUpdated(ctx context.Context, cli client.Client) (bool, error) {
```

Comment:
should this function also check that the pod is ready?

Reply:
added